### ISSUE TITLE
Combining build settings with inclusive build tag filters causes Bazel to crash

### ISSUE THREAD

**Issue by jonatanj** [bazelbuild/bazel#10213]
### Description of the problem:

I'm trying to build a subset of targets by using the --build-tag-filters option while using a custom build setting. However, using the minimal example below, the following build command causes Bazel to crash with a backtrace:

```
$ bazel --ignore_all_rc_files build --//:foo=True --build_tag_filters=some_tag //...
INFO: Writing tracer profile to '/home/jonatan/.cache/bazel/_bazel_jonatan/c723c8a0fd8fe927056d57db06e6b571/command.profile.gz'
Internal error thrown during build. Printing stack trace: java.util.NoSuchElementException
	at com.google.common.collect.AbstractIndexedListIterator.next(AbstractIndexedListIterator.java:75)
	at com.google.common.collect.Iterators.getOnlyElement(Iterators.java:302)
	at com.google.common.collect.Iterables.getOnlyElement(Iterables.java:254)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.loadBuildSetting(StarlarkOptionsParser.java:212)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.parseArg(StarlarkOptionsParser.java:162)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.parse(StarlarkOptionsParser.java:101)
	at com.google.devtools.build.lib.runtime.BlazeOptionHandler.parseStarlarkOptions(BlazeOptionHandler.java:250)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:527)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:596)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:647)
	at java.base/java.lang.Thread.run(Unknown Source)
java.util.NoSuchElementException
	at com.google.common.collect.AbstractIndexedListIterator.next(AbstractIndexedListIterator.java:75)
	at com.google.common.collect.Iterators.getOnlyElement(Iterators.java:302)
	at com.google.common.collect.Iterables.getOnlyElement(Iterables.java:254)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.loadBuildSetting(StarlarkOptionsParser.java:212)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.parseArg(StarlarkOptionsParser.java:162)
	at com.google.devtools.build.lib.runtime.StarlarkOptionsParser.parse(StarlarkOptionsParser.java:101)
	at com.google.devtools.build.lib.runtime.BlazeOptionHandler.parseStarlarkOptions(BlazeOptionHandler.java:250)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:527)
	at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:204)
	at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:596)
	at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:647)
	at java.base/java.lang.Thread.run(Unknown Source)
WARNING: Waiting for server process to terminate (waited 5 seconds, waiting at most 60)
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

```
cat >WORKSPACE <<EOF
load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")

http_archive(
    name = "bazel_skylib",
    url = "https://github.com/bazelbuild/bazel-skylib/releases/download/0.9.0/bazel_skylib-0.9.0.tar.gz",
    sha256 = "1dde365491125a3db70731e25658dfdd3bc5dbdfd11b840b3e987ecf043c7ca0",
)
EOF

cat >BUILD <<EOF
load("@bazel_skylib//rules:common_settings.bzl", "bool_flag")

bool_flag(
    name = "foo",
    build_setting_default = False,
)

config_setting(
    name = "foo_config_setting",
    flag_values = {":foo": "True"}
)

filegroup(
    name = "bar",
    srcs = select({
        ":foo_config_setting": [],
        "//conditions:default": []
    }),
    tags = ["some_tag"]
)
EOF
```

### What operating system are you running Bazel on?

Ubuntu 18.04.1

### What's the output of `bazel info release`?

release 1.1.0

###  Have you found anything relevant by searching the web?

No.

### Any other information, logs, or outputs that you want to share?

It might be worth noting the results of some similar commands:

Excluding a tag using "-": OK
`bazel build --//:foo=True --test_tag_filters=-some_tag //...`

Using --test_tag_filters: OK
`bazel test --//:foo=True --test_tag_filters=some_tag //...`

Using a non-existing tag: fails with stracktrace
`bazel build --//:foo=True --build_tag_filters=some_non_existing_tag //...`

Using just the build setting: OK
`bazel build --//:foo=True //...`

Using just the inclusive build tag filter: OK
`bazel build --build_tag_filters=some_tag //...`

**Comment by katre** [comment]
Is the build tag filter causing the flag to not be built? Ugh. Will investigate.

**Comment by benjaminp** [comment]
This was probably fixed by 8fd43cf2effa3c1727fb2fb63390d7db855dc5ad.

**Comment by katre** [comment]
The actual crash seems to be fixed now (thanks for flagging 8fd43cf2effa3c1727fb2fb63390d7db855dc5ad, @benjaminp).
